src/FormNetworks/bilinearformnetwork.jl

```diff
@@ -0,0 +1,62 @@
+default_bra_vertex_map(v) = (v, "bra")
+default_ket_vertex_map(v) = (v, "ket")
+default_operator_vertex_map(v) = (v, "operator")
+default_operator_constructor(s::IndsNetwork) = delta_network(s)
+
+struct BilinearFormNetwork{V,KetMap,BraMap,OperatorMap} <: AbstractFormNetwork{V}
+  tn::AbstractITensorNetwork{V}
+  bra_vertex_map::BraMap
+  ket_vertex_map::KetMap
+  operator_vertex_map::OperatorMap
+end
+
+function BilinearFormNetwork(
+  operator::AbstractITensorNetwork,
+  bra::AbstractITensorNetwork,
+  ket::AbstractITensorNetwork;
+  bra_vertex_map=default_bra_vertex_map,
+  ket_vertex_map=default_ket_vertex_map,
+  operator_vertex_map=default_operator_vertex_map,
+)
+  @assert Set(externalinds(operator)) ==
+    union(Set(externalinds(ket)), Set(externalinds(bra)))
+  @assert isempty(findall(in(internalinds(bra)), internalinds(ket)))
+  @assert isempty(findall(in(internalinds(bra)), internalinds(operator)))
+  @assert isempty(findall(in(internalinds(ket)), internalinds(operator)))
+
+  bra_renamed = rename_vertices_itn(bra, bra_vertex_map)
+  ket_renamed = rename_vertices_itn(ket, ket_vertex_map)
+  operator_renamed = rename_vertices_itn(operator, operator_vertex_map)
+
+  tn = union(union(bra_renamed, operator_renamed), ket_renamed)
```

Comment:
```suggestion
  # TODO: Reminder to fix `union` so that `union(bra_renamed, operator_renamed, ket_renamed)` works.
  tn = union(union(bra_renamed, operator_renamed), ket_renamed)
```